### ISSUE TITLE
Add redirect tests to Bedrock

### ISSUE THREAD

**Issue by rbillings** [mozilla/mcom-tests#434]
Copy redirect tests into Bedrock repo

**Comment by davehunt** [comment]
Could you provide more details @rbillings? Is the plan to ultimately move these tests into the bedrock repository?

**Comment by rbillings** [comment]
This came from a discussion with the Bedrock team. Because testing redirects is a high priority for the move, they want a copy of these tests in their repo. So it isn't a removal so much as an addition to their tests.

**Comment by rbillings** [comment]
And with further discussion it makes more sense to move them into the repo instead of having them be two places.

**Comment by davehunt** [comment]
I'm happy to help with this. Let's try to gather the details of the tests to move and identify a suitable destination in this issue. Almost certainly something within https://github.com/mozilla/bedrock/tree/master/tests. It could make sense to move all http(s) checks, not just redirect. By this I mean all tests that are not using a browser, and are simply checking URLs are returning the appropriate responses.

**Comment by davehunt** [comment]
We'll also need to update Jenkins to get the tests from the bedrock repository, once this is complete.

**Comment by davehunt** [comment]
I just noticed https://github.com/pmclanahan/mozorg-redirects mentioned on IRC. We should probably understand who's working on this so we don't duplicate efforts. /cc @pmclanahan

**Comment by pmclanahan** [comment]
Hi! Indeed I'm working on a redirect specific test suite as part of our move of redirects in apache configs into bedrock itself. The current work is in the bedrock repo itself in mozilla/bedrock#3251. After that merges we can combine that suite with what HTTP tests are in this repo, or move this over as a separate thing.

That said I don't necessarily think that these things overlap since my work is around double checking that bedrock specifically is handling these redirects, but they will likely end up checking some of the same URLs.

**Comment by jgmize** [comment]
@pmclanahan now that we have the redirect tests in bedrock, do you believe we're at the point where all of the redirect test coverage in this repo is redundant and can be safely removed in favor of running the bedrock redirect tests as a separate job in jenkins?

As a separate issue, which I was going to file against this repo but in light of this issue probably makes more sense in a PR against bedrock, I believe we should tag redirects to external sites separately from internal redirects and run them in a separate jenkins job, similar to how we treat link tests now.

**Comment by pmclanahan** [comment]
We'd have to audit the redirect tests to ensure we have coverage in the other suite. A major difference in the bedrock redirects suite and this one is that the bedrock tests don't check anything but the response from bedrock. The mcom-tests assert a final url for redirection and make sure the destination urls work. I'm sure there are good candidates for this kind of testing, but the concern of the tests in bedrock is that bedrock is acting as we expect and asserts nothing about the destination in the returned `Location:` header. So we may need to keep some of the mcom-tests redirect tests if we do want to assert something about the destination URLs. I don't want those in the new redirects test suite in bedrock as it would significantly slow them as well as make them prone to failure when another site is down or changes.

**Comment by jgmize** [comment]
Yes, I agree that we should keep tests that assert something about the redirect target separate, especially for cases where that target is a different site. However, separation can be achieved without maintaining separate repositories, and I'd like to continue our current path of consolidation into the bedrock repo, so that changes to the tests can be part of the same PR that changes behavior. What if we copied the tests in [test_redirect.py](https://github.com/mozilla/mcom-tests/blob/master/tests/test_redirect.py) and [test_redirect_landing.py](https://github.com/mozilla/mcom-tests/blob/master/tests/test_redirect.py) over to the bedrock repo in the redirect_tests dir with some markers to differentiate them from the redirect tests that only check the location header, and we exclude tests with that marker from running in travis or the deployment pipeline. Then, we can run the redirect tests that actually hit external destinations together with the link tests that hit external sites, and maybe add some logic so that we get failures for 40x response codes but only warnings for 50x response codes under the assumption that we don't need to take action for temporary issues on another site.

**Comment by davehunt** [comment]
My initial attempt is here: https://github.com/mozilla/bedrock/pull/3785

**Comment by davehunt** [comment]
This has been completed.
